### PR TITLE
Simplify and clarify pubsub registry interactions

### PR DESCRIPTION
**Pull request by jessemyers**
Main changes:
 -  Clearly differentiate convention-driven and class-driven schema registration
 -  Handle cases with multiple daemons registering different handlers for the same media type
 -  Make fixtures more obvious

### CODE REVIEW

File: microcosm_pubsub/daemon.py [hunk 1]
```diff
@@ -9,18 +9,30 @@
...
+        self.bound_handlers = None
+
     def make_arg_parser(self):
         parser = super(ConsumerDaemon, self).make_arg_parser()
         parser.add_argument("--sqs-queue-url")
         return parser
 
+    def create_object_graph_components(self, graph):
+        super(ConsumerDaemon, self).create_object_graph_components(graph)
+        self.bound_handlers = graph.sqs_message_handler_registry.compute_bound_handlers(
```
Comment (jessemyers) [inline microcosm_pubsub/daemon.py]: It's possible to have multiple daemons in the same code base; in this case, the handler registry could have more than one entry. We solve this by determining which handlers are part of the daemon's object graph and restricting access to just those handlers.

File: microcosm_pubsub/tests/fixtures.py [hunk 2]
```diff
@@ -4,67 +4,64 @@
...
-        return "example"
-
-
-class Foo(object):
-    pass
-
-
 @schema
-class FooSchema(PubSubMessageSchema):
+class DerivedSchema(PubSubMessageSchema):
```
Comment (jessemyers) [inline microcosm_pubsub/tests/fixtures.py]: I tried to get rid of `foo` and `bar` and be more explicit with my naming. This avoids some spurious test cases where we have duplicate names.

File: microcosm_pubsub/daemon.py [hunk 3]
```diff
@@ -9,18 +9,30 @@
...
+            self.components,
+        )
+
     def run_state_machine(self):
-        for media_type in self.graph.sqs_message_handler_registry.keys():
-            handler = self.graph.sqs_message_handler_registry.find(media_type)
+        self.compute_bound_handlers()
```
Comment (netama) [inline microcosm_pubsub/daemon.py]: where is this defined?

**Comment by netama** [comment]
LGTM with one question

File: microcosm_pubsub/daemon.py (hunk 3 above)
Comment (jessemyers) [inline microcosm_pubsub/daemon.py]: oops, that's from before a refactor

**Pull request closed without merging**